File: frontend/src/api/ai.ts
const BASE_URL = "http://YOUR_LAPTOP_IP:8080";

export async function askAI(message: string) {
  const res = await fetch(`${BASE_URL}/ai/chat`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ message }),
  });

  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(err?.error || "AI request failed");
  }

  return res.json() as Promise<{ reply: string }>;
}

File: backend/routes/ai.go
package routes

import (
	"context"
	"os"

	"github.com/gofiber/fiber/v3"
	openai "github.com/openai/openai-go"
)

type AIReq struct {
	Message string `json:"message"`
}

func RegisterAIRoutes(app *fiber.App) {
	client := openai.NewClient(
		openai.WithAPIKey(os.Getenv("OPENAI_API_KEY")),
	)

	app.Post("/ai/chat", func(c fiber.Ctx) error {
		var body AIReq
		if err := c.Bind().Body(&body); err != nil {
			return c.Status(400).JSON(fiber.Map{"error": "invalid body"})
		}
		if body.Message == "" {
			return c.Status(400).JSON(fiber.Map{"error": "message required"})
		}

		resp, err := client.Chat.Completions.New(context.Background(), openai.ChatCompletionNewParams{
			Model: openai.F("gpt-4o-mini"),
			Messages: openai.F([]openai.ChatCompletionMessageParamUnion{
				openai.SystemMessage("You are a concise shopping assistant for this store. Do not invent products."),
				openai.UserMessage(body.Message),
			}),
		})
		if err != nil {
			return c.Status(500).JSON(fiber.Map{"error": "openai request failed"})
		}

		return c.JSON(fiber.Map{
			"reply": resp.Choices[0].Message.Content,
		})
	})
}
